### DISCUSSION TITLE
SQLite instead of MariaDB

### DISCUSSION THREAD

**Discussion by briped** [hassio-addons/addon-nginx-proxy-manager#255] [Ideas]
Would it be possible to use the SQLite integration rather than MariaDB? That way the addon/container wouldn't be dependent on another addon.

Looks like it's "just" an environment variable that needs to be added (and all MYSQL variables removed). I don't know if it could be made in a way so the option is configurable in the addon?

**Comment by frenck** [comment]
I would love to make it the default, but that would be massive, as in, problematic with handling the add-on update push

**Reply by briped** [reply]
Thank you for the fast reply. I had hoped it was an easy change, just shows how little I know about HassOS and addons :)

For now, I've "installed" NPM with SQLite using Portainer. I might just go with the MariaDB version, just to have everything consolidated under the HassOS Supervisor :)

**Comment by iBobik** [comment]
@herrfrei Would you recommend to use your fork? Are you going to maintain it?
https://github.com/herrfrei/addon-nginx-proxy-manager/tree/main

New users does not need migration, so no need for big changes to make it publicly useable.

**Reply by herrfrei** [reply]
At the moment I do not use it actively. Also works only with a new installation (no automatic migration). I just updated the fork to the current state, please test it if you want!

If you need to migrate, you might try the following steps:

1.    Deploy the addon and stop it once it ihas started
2.    Get the database.sqlite from the fresh instance
3.    Migrate from MySQL/MariaDB (e.g. via DBeaver)
       - Don't choose "Truncate target tables" (leave unchecked!)
       - Migrate
4. Put back the SQLite database to the addon
5.    Start Nginx addon again

**Comment by frenck** [comment]
The current development version is using SQLite and will be shipped in the next release of the add-on.

../Frenck

**Reply by Techknowledgeman** [reply]
Only problem with that is you've broken MySQL installs. How can I force it back to mariadb?
